main/test/ca/uwaterloo/flix/language/phase/TestResolver.scala

```diff
@@ -295,6 +295,24 @@ class TestResolver extends FunSuite with TestUtils {
     expectError[ResolutionError.UndefinedName](result)
   }
 
+  test("UndefinedName.04") {
+    val input =
+      """
+        |instance C[Int]
+        |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[ResolutionError.UndefinedName](result)
+  }
+
+  test("UndefinedName.05") {
+    val input =
+      """
+        |def f[a: C](x: a): a = x
+        |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[ResolutionError.UndefinedName](result)
```

Comment:
It seems more like a redundancy error, right? Eventually I'd like to consider superclasses such that `[a: C, SuperC]` would also be an error.

Reply:
It could also be a redundancy error, but if it is purely syntactic then usually it is handled in the Weeder. But given that you have more complex scenarios, maybe we should leave it for the Redundancy phase.